File: python/Mahasiswa.py
# Membuat kelas baru bernama Mahasiswa, class ini merupakan child class dari SivitasAkademik Class
from SivitasAkademik import SivitasAkademik

class Mahasiswa(SivitasAkademik):
    
    # Inisialisasi variabel untuk setiap objek dari kelas
    nim = ""        # Membuat variabel untuk NIM
    fakultas = ""   # Membuat variabel untuk fakultas
    
    # Konstruktor untuk inisialisasi variabel pada objek dari kelas Mahasiswa
    def __init__(self, NIK = "", name = "", gender = "", asal_univ = "", email_edu = "", nim = "", fakultas = ""):
        super().__init__(NIK, name, gender, asal_univ, email_edu)
        self.nim = nim              # Menginisialisasi variabel nim
        self.fakultas = fakultas    # Menginisialisasi variabel fakultas
        
    # Getter dan Setter untuk NIM
    def get_Nim(self):
        return self.nim
    
    def set_Nim(self, nim):
        self.nim = nim
        
    # Getter dan Setter untuk Fakultas
    def get_Fakultas(self):
        return self.fakultas
    
    def set_Fakultas(self, fakultas):
        self.fakultas = fakultas

File: cpp/Mahasiswa.cpp
class Mahasiswa :  public SivitasAkademik{
    
    /*
    Saya Talitha Syahla NIM 2101330 mengerjakan Soal Latihan 4 
    dalam mata kuliah Desain Pemrograman Berorientasi Objek 
    untuk keberkahanNya maka saya tidak melakukan kecurangan 
    seperti yang telah dispesifikasikan. Aamiin.
    */

    private:
        string nim;
        string fakultas;

    public:
    // Constructor
        Mahasiswa(){
            this->nim = "";
            this->fakultas = "";
        }

        Mahasiswa(string NIK, string name, string gender, string asal_univ, string email_edu,string nim, string fakultas) : SivitasAkademik(NIK, name, gender, asal_univ, email_edu){
            this->nim = nim;
            this->fakultas = fakultas;
        }

    // Setter dan Getter

        // Untuk NIM
        string get_Nim(){
            return this->nim;
        }

        void set_Nim(string nim){
            this->nim = nim;
        }

        // Untuk Fakultas
        string get_Fakultas(){
            return this->fakultas;
        }

        void set_Fakultas(string fakultas){
            this->fakultas = fakultas;
        }

        // Destruktor
        ~Mahasiswa(){
        }
    
};